### ISSUE TITLE
Xcode has no way pass env variables to react-native server

### ISSUE THREAD

**Issue by chrmod** [facebook/react-native#27709]
## Description:

To start react-native server XCode has to run:
* `launchPackager.command` which starts new terminal emulator window
* which runs `packager.sh`, this process as it runs in new terminal has no access to environment variables
* to pass variables `.packager.env` file is created by XCode build script 
* `.packager.env` **was** sourced before starting react-native server https://github.com/facebook/react-native/commit/9ece5bda9bd035c31c3b975d1a8b66574dd43c4e#diff-fe2ea122979e8bb9729dff893828ae61L14

Our use case is to pass a path to another `NODE_BINARY` as we want to run exact version with help of `nodenv` (or nvm).

It seems that before https://github.com/facebook/react-native/commit/9ece5bda9bd035c31c3b975d1a8b66574dd43c4e#diff-fe2ea122979e8bb9729dff893828ae61L14 we could just pass `NODE_BINARY` to `.packager.env`.

My assumtion is that `RCT_METRO_PORT` is now ignored and thus port cannot be customized.

## React Native version:

```
  System:
    OS: macOS 10.15.3
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 2.40 GB / 32.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 10.17.0 - ~/.nvm/versions/node/v10.17.0/bin/node
    npm: 6.11.3 - ~/.nvm/versions/node/v10.17.0/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
  IDEs:
    Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
  npmPackages:
    react: ^16.9.0 => 16.9.0
    react-native: ^0.61.5 => 0.61.5
  npmGlobalPackages:
    react-native-cli: 2.0.1
```

## Expected Results

While building with XCode, build steps should be able to pass environment variables to configure react-native server.

**Comment by chrmod** [comment]
cc @hramos

**Comment by chrmod** [comment]
@hramos what kind of information you would like get? How can we fix this? Would like accept a patch?

**Comment by hramos** [comment]
@chrmod can you edit your issue and make sure it uses the template provided? It's the block of text that is prepopulated when you open a new issue. PRs always welcome.

**Comment by chrmod** [comment]
@hramos is this better?
